internal/rbd/controllerserver.go

```diff
@@ -1416,8 +1422,8 @@ func (cs *ControllerServer) DeleteSnapshot(
 	}
 	defer cs.OperationLocks.ReleaseDeleteLock(snapshotID)
 
-	rbdSnap := &rbdSnapshot{}
-	if err = genSnapFromSnapID(ctx, rbdSnap, snapshotID, cr, req.GetSecrets()); err != nil {
+	rbdSnap, err := genSnapFromSnapID(ctx, snapshotID, cr, req.GetSecrets())
```

Comment:
in case of err , rbdSnap is still used in `cleanUpImageAndSnapReservation` which is causing issues.

Reply:
yes, I noticed. Super ugly :sob:

Reply:
Tested manually with returning a partial generated rbdSnapshot for now. It is not as clean as I would like, but it is still better than what was done before.